src/utils/Configuration.ts

```diff
@@ -121,6 +125,35 @@ class Configuration {
       return process.env.SECRET_KEY;
     }
   }
+
+  /**
+   * Method to get the secret details for the Welsh lookup
+   * @returns ISecret secret containing SMC url and api key
+   */
+  public async getSecret() {
+    const welshConfigSecretKey: string = this.getWelshSecretKey();
+
+    if (welshConfigSecretKey) {
+      try {
+        const secretRequest: GetSecretValueRequest = {SecretId: welshConfigSecretKey};
+        const secretResponse: GetSecretValueResponse = await this.secretsClient.getSecretValue(secretRequest).promise();
+
+        if (secretResponse.SecretString) {
+          const secretConfig: ISecret = JSON.parse(secretResponse.SecretString);
+          return secretConfig;
+        } else {
+          console.log(ERRORS.SECRET_DETAILS_NOT_FOUND);
+          return null;
+        }
+      } catch (error) {
+        console.log(error);
+        return null;
+      }
+    } else {
+      console.log(ERRORS.SECRET_ENV_VAR_NOT_EXIST);
+      return null;
+    }
+  }
```

Comment:
All I have done is move this code from the Service into Configuration, so this was already there. However I do like the suggestion of adding the try/catch, will also add return type.

With regards to the async / await removal, I would be very reluctant to remove that. The logic is consistent with all of our other services (where there have been zero issues with regards to Secret retrieval), when promisifying the getSecretValue response and awaiting for the GetSecretValueResponse. In my mind it seems the right thing to do, as we would want make sure the response has resolved before retrieving the SecretString from it would't we?

With regards to the ternary / one-liner changes, call me old fashioned but I think the current way is more readable.